iocore/net/UnixUDPNet.cc

```diff
@@ -628,6 +666,24 @@ UDPNetProcessor::CreateUDPSocket(int *resfd, sockaddr const *remote_addr, Action
     }
   }
 
+  if (opt.ip_family == AF_INET) {
+    int enable = 1;
+#ifdef IP_PKTINFO
+    safe_setsockopt(fd, IPPROTO_IP, IP_PKTINFO, reinterpret_cast<char *>(&enable), sizeof(enable));
```

Comment:
nikpick: should we continue when setsockopt failed ?

Reply:
Added error checks. It's complicated a bit, because it fails even if `IP_PKTINFO` is defined and try the other to support various platforms.